e2e/tests/test_cases/single_replica_node_down.robot

```diff
@@ -0,0 +1,118 @@
+
+*** Variables ***
+${LOOP_COUNT}    1
+${RETRY_COUNT}    300
+${RETRY_INTERVAL}    1
+${SHUTDOWN_WAIT_DURATION}    180
+
+*** Test Cases ***
+Single Replica Node Down Deletion Policy do-nothing With RWO Volume Data Locality disabled
+    When Create persistentvolumeclaim 0 using RWO volume
+    And Create deployment 0 with persistentvolumeclaim 0
+    And Write 100 MB data to file data in deployment 0
+    And Update persistentvolumeclaim 0 volume replica count to ${1}
+    And Delete replica of deployment 0 volume on replica node
+    And Delete replica of deployment 0 volume on volume node
+    And Power off volume node of deployment 0
+    And Sleep   ${SHUTDOWN_WAIT_DURATION}
+    Then Wait for volume of deployment 0 attaching
+
+    When Power on volume node of deployment 0
+    And Wait for deployment 0 pods stable
+    Then Check deployment 0 data in file data is intact
+
+Single Replica Node Down Deletion Policy do-nothing With RWO Volume Data Locality
+    When Create persistentvolumeclaim 0 using RWO volume
+    And Create deployment 0 with persistentvolumeclaim 0
+    And Write 100 MB data to file data in deployment 0
+    And Update persistentvolumeclaim 0 volume replica count to ${1}
+    And Delete replica of deployment 0 volume on all replica node
+    And Power off volume node of deployment 0
+    And Sleep   ${SHUTDOWN_WAIT_DURATION}
+    Then Wait for volume of deployment 0 faulted
+
+    When Power on off nodes
+    And Wait for deployment 0 pods stable
+    Then Check deployment 0 data in file data is intact
+
+Single Replica Node Down Deletion Policy delete-deployment-pod With RWO Volume Data Locality disabled
+    Given Set setting node-down-pod-deletion-policy to delete-deployment-pod
+
+    When Create persistentvolumeclaim 0 using RWO volume
+    And Create deployment 0 with persistentvolumeclaim 0
+    And Write 100 MB data to file data in deployment 0
+    And Update persistentvolumeclaim 0 volume replica count to ${1}
+    And Delete replica of deployment 0 volume on replica node
+    And Delete replica of deployment 0 volume on volume node
+    And Power off volume node of deployment 0
+    And Sleep   ${SHUTDOWN_WAIT_DURATION}
+    Then Wait for volume of deployment 0 attaching
+
+    And Wait for deployment 0 pods stable
+    Then Check deployment 0 data in file data is intact
+
+Single Replica Node Down Deletion Policy delete-deployment-pod With RWO Volume Data Locality
+    Given Set setting node-down-pod-deletion-policy to delete-deployment-pod
+
+    When Create persistentvolumeclaim 0 using RWO volume
+    And Create deployment 0 with persistentvolumeclaim 0
+    And Write 100 MB data to file data in deployment 0
+    And Update persistentvolumeclaim 0 volume replica count to ${1}
+    And Delete replica of deployment 0 volume on all replica node
+    And Power off volume node of deployment 0
+    And Sleep   ${SHUTDOWN_WAIT_DURATION}
+    Then Wait for volume of deployment 0 faulted
```

Comment:
According to the original test case, seems we have to check the following as well?
```
Wait for pod deletion & recreation on another node. The pod recreation will not happen immediately.
The replacement workload pod will get stuck in the ContainerCreating state.
```

Reply:
Updated